tests/pages.ts

```diff
@@ -79,21 +82,80 @@ export const workspaces = {
 
 export const newWorkspace = {
     goTo: (page: Page, contextUrl = 'https://github.com/gitpod-io/empty') => page.goto(`https://${process.env.GITPOD_HOST}/new#${contextUrl}`),
-    continue: async (page: Page) => {
+    continue: async (page: Page, workspaceClass?: WorkspaceClass, errorMsg?: string) => {
+        if (workspaceClass) {
+            await workspaceClassDropDown.set(page, workspaceClass, true);
+        }
         await page.getByRole('button', { name: 'Continue' }).click();
+        if (errorMsg) {
+            await expect(page.getByText(errorMsg)).toBeVisible();
+            return
+        }
         await page.waitForURL(`https://${process.env.GITPOD_HOST}/start/**`);
+    },
+    Errors: {
+        ClassNotAllowed: 'workspace class is not allowed',
     }
 }
 
+export const orgSettings = {
+    goTo: (page: Page) => page.goto(`https://${process.env.GITPOD_HOST}/settings`),
+    setWorkspaceShared: async (page: Page, shared: boolean) => {
+        await page.getByLabel('Workspace SharingAllow').click();
+        expect(await page.getByLabel('Workspace SharingAllow').isChecked()).toBe(shared);
+    },
+    checkWorkspaceClass: async (page: Page, id: WorkspaceClass, checked: boolean) => {
+        const isChecked = await page.locator('#' + id).isChecked();
+        if ((isChecked && checked) || (!isChecked && !checked)) {
+            return;
+        }
+        const checkBox = page.locator("#" + id)
+        if (checked) {
+            await checkBox.check()
+        } else {
+            await checkBox.click();
+        }
+    },
+    setAllowedWorkspaceClasses: async (page: Page, classes: WorkspaceClass[]) => {
+        await orgSettings.goTo(page);
+        
+        await Promise.all(AllWorkspaceClass.filter(e => classes.includes(e)).map(e => orgSettings.checkWorkspaceClass(page, e, true)))
+        await Promise.all(AllWorkspaceClass.filter(e => !classes.includes(e)).map((e, i) => orgSettings.checkWorkspaceClass(page, e, false)))
+        if (classes.length === 0) {
+            await page.locator('div').filter({ hasText: /^Should have one workspace class selected at least\.$/ }).isVisible();
```

Comment:
don't use such locator

use `expect(page.getText('/^Should have one workspace class selected at least\.$/)).toBeVisibile()`

it relies on rendered text and waits properly